server/src/test/java/com/objectcomputing/checkins/services/file/FileServicesImplTest.java

```diff
@@ -0,0 +1,1019 @@
+import org.mockito.stubbing.Answer;
+
+import java.io.*;
+import java.time.LocalDate;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@MicronautTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class FileServicesImplTest {
+
+    private static File testFile;
+    private final static String filePath = "testFile.txt";
+    final static JsonFactory jsonFactory = new MockJsonFactory();
+
+    @Mock
+    private Authentication authentication;
+
+    @Mock
+    private Map mockAttributes;
+
+    @Mock
+    private Drive drive;
+
+    @Mock
+    private Drive.Files files;
+
+    @Mock
+    private Drive.Files.List list;
+
+    @Mock
+    private Drive.Files.Get get;
+
+    @Mock
+    private Drive.Files.Delete delete;
+
+    @Mock
+    private Drive.Files.Create create;
+
+    @Mock
+    private CompletedFileUpload fileToUpload;
+
+    @Mock
+    private MemberProfile testMemberProfile;
+
+    @Mock
+    private CheckIn testCheckIn;
+
+    @Mock
+    private CheckinDocument testCd;
+
+    @Mock
+    private InputStream mockInputStream;
+
+    @Mock
+    private CheckInServices checkInServices;
+
+    @Mock
+    private CheckinDocumentServices checkinDocumentServices;
+
+    @Mock
+    private GoogleDriveAccessor googleDriveAccessor;
+
+    @Mock
+    private SecurityService securityService;
+
+    @Mock
+    private CurrentUserServices currentUserServices;
+
+    @Mock
+    private MemberProfileServices memberProfileServices;
+
+    @Mock
+    private CompletedFileUpload completedFileUpload;
+
+    @InjectMocks
+    private FileServicesImpl services;
+
+    @BeforeAll
+    void initMocksAndInitializeFile() throws IOException {
+        MockitoAnnotations.initMocks(this);
+
+        testFile = new File(filePath);
+        FileWriter myWriter = new FileWriter(testFile);
+        myWriter.write("This.Is.A.Test.File");
+        myWriter.close();
+    }
+
+    @AfterAll
+    void deleteTestFile() {
+        testFile.deleteOnExit();
+    }
+
+    @BeforeEach
+    void resetMocks() {
+        Mockito.reset(authentication);
+        Mockito.reset(mockAttributes);
+        Mockito.reset(drive);
+        Mockito.reset(files);
+        Mockito.reset(list);
+        Mockito.reset(get);
+        Mockito.reset(delete);
+        Mockito.reset(create);
+        Mockito.reset(fileToUpload);
+        Mockito.reset(testMemberProfile);
+        Mockito.reset(testCheckIn);
+        Mockito.reset(testCd);
+        Mockito.reset(mockInputStream);
+        Mockito.reset(checkInServices);
+        Mockito.reset(checkinDocumentServices);
+        Mockito.reset(googleDriveAccessor);
+        Mockito.reset(securityService);
+        Mockito.reset(currentUserServices);
+        Mockito.reset(memberProfileServices);
+        Mockito.reset(completedFileUpload);
+    }
+
+    private void setupMocksForAuth() {
```

Comment:
This is in every test. Would it make sense to put it in the BeforeEach?

Reply:
Good catch, not sure how I missed it. I've moved it to beforeEach now.